tests/pipeline/test_pipeline.py

```diff
@@ -791,6 +791,65 @@ def test_only_nodes_unknown(self, pipeline_list_with_lists, target_node_names):
         with pytest.raises(ValueError, match=pattern):
             full.only_nodes(*target_node_names)
 
+    @pytest.mark.parametrize(
+        "non_namespaced_node_name",
+        ["node1", "node2", "node3", "node4", "node5", "node6"],
+    )
+    def test_only_nodes_with_namespacing(
+        self, pipeline_with_namespaces, non_namespaced_node_name
+    ):
+        # Tests that error message will supply correct namespaces.
+        # Example of expected error:
+        # Pipeline does not contain nodes named ['node1']. Did you mean: ['katie.node1']?
+        pattern = (
+            rf"Pipeline does not contain nodes named \['{non_namespaced_node_name}'\]\. "
+            rf"Did you mean: \['.*\.{non_namespaced_node_name}'\]\?"
+        )
+        full = pipeline_with_namespaces
+        with pytest.raises(ValueError, match=pattern):
+            full.only_nodes(non_namespaced_node_name)
+
+    @pytest.mark.parametrize(
+        "non_namespaced_node_names",
+        [("node1", "node2"), ("node3", "node4"), ("node5", "node6")],
+    )
+    def test_only_nodes_with_namespacing_multiple_args(
+        self, pipeline_with_namespaces, non_namespaced_node_names
+    ):
+        # Tests that error message will contain suggestions for all provided arguments.
+        # Example of expected error message:
+        # "Pipeline does not contain nodes named ['node1', 'node2'].
+        # Did you mean: ['katie.node1', 'lisa.node2']?"
```

Comment:
These comments help a lot in understanding the tests, but I'd be in favour of refactoring some of this to make it more readable in a separate task. Can you create an issue for that?